docs/src/app/docs/customization/page.mdx

```diff
@@ -61,3 +61,44 @@ export const env = createEnv({
   isServer: typeof window === "undefined",
 });
 ```
+
+## Extending Other Envs
+
+In some case, different parts of your application may depend on different environment variables. This is especially common in monorepos. Using the `extends` property, you can extend another `t3-env` instance with additional variables:
+
+```ts title="packages/env/baseEnv.ts"
+import { createEnv } from "@t3-oss/env-core";
+
+export const env = createEnv({
+  shared: {
+    DEBUG: z.enum(["true", "false"]).transform((value) => value === "true"),
+  },
+  server: {
+    NODE_ENV: z.enum(["development", "production"]),
+  },
+});
+```
+
+```ts title="packages/db/src/env.ts"
+import { createEnv } from "@t3-oss/env-core";
+import { env as baseEnv } from "my-monorepo/env/baseEnv";
+
+export const env = createEnv({
+  extends: baseEnv,
+  server: {
+    DB_HOST: z.string(),
+    DB_PORT: z.number(),
+  },
+});
+
+// Access variables from the extended env just like you would otherwise:
+if (env.DEBUG) {
+  console.log(`NODE_ENV: ${env.NODE_ENV}, DB_HOST: ${env.DB_HOST}`);
+}
+```
+
+<Callout type="warning">
+
+The extending env does not inherit any of the other env's options, such as `clientPrefix`, `emptyStringAsUndefined`, `runtimeEnv`, etc. You need to define these separately in each env.
```

Comment:
When does this even matter? It should only use the outer most config option right? The presets ideally shouldn't have to pass any options like this if presets should be truly reusable without side-effects?